### ISSUE TITLE
IAM Instance Profile Change - Replacement Required

### ISSUE THREAD

**Issue by moofish32** [hashicorp/terraform#8924]
Hi there,
### Terraform Version

Terraform v0.7.3
### Affected Resource(s)

Please list the resources as a list, for example:
- aws_instance
- iam_instance_profile
### Terraform Configuration Files

``` hcl
variable "region"        { default = "us-west-2" }
variable "key_name"      { }
variable "aws_profile"   { }
variable "asg_desired"   { default = 1 }
variable "asg_max"       { default = 1 }
variable "asg_min"       { default = 1 }
variable "instance_name" { default = "iam_profile_bug"}
variable "ami_id"        { }
variable "service_tag"   { default = "iam_profile_bug" }
variable "instance_type" { default = "t2.micro" }
variable "tenancy"       { default = "default" }

variable "bastion_ref_sec_grp" { } 
variable "app_subnets"         { type = "list" default = [""] }

provider "aws" {
  region  = "${var.region}"
  profile = "${var.aws_profile}"
}

resource "aws_iam_instance_profile" "eip_profile" {
  lifecycle{ create_before_destroy = true }
  name_prefix = "eip_profile"
  roles = ["${aws_iam_role.eip_role.name}"]
}

resource "aws_iam_role" "eip_role" {
  name_prefix = "eip_role-"
  path = "/"
  lifecycle{ create_before_destroy = true }
  assume_role_policy = <<EOF
{
    "Statement": [
      {
        "Action": [
          "sts:AssumeRole"
        ],
        "Effect": "Allow",
        "Principal": {
          "Service": [ "ec2.amazonaws.com" ]
        }
      }
    ],
    "Version": "2012-10-17"
}
EOF
}

resource "aws_iam_role_policy" "eip_policy" {
  name   = "eip_policy"
  role   = "${aws_iam_role.eip_role.id}"
  policy = <<EOF
{
  "Statement": [
    {
      "Action": [
        "ec2:DescribeAddresses",
        "ec2:AssociateAddress"
      ],
      "Effect": "Allow",
      "Resource": "*"
    }
  ],
  "Version": "2012-10-17"
}
EOF
}

resource "aws_launch_configuration" "iam_profile_bug" {
  lifecycle { create_before_destroy = true }

  name_prefix   = "bug-launch-config-"
  image_id      = "${var.ami_id}"
  instance_type = "${var.instance_type}"

  key_name             = "${var.key_name}"
  placement_tenancy    = "${var.tenancy}"
  security_groups      = ["${var.bastion_ref_sec_grp}"]
  /* iam_instance_profile = "${aws_iam_instance_profile.eip_profile.name}" */
}

resource "aws_autoscaling_group" "iam_bug" { 
  lifecycle { create_before_destroy = true }

  vpc_zone_identifier       = "${var.app_subnets}"
  desired_capacity          = "${var.asg_desired}"
  max_size                  = "${var.asg_max}"
  min_size                  = "${var.asg_min}"
  launch_configuration      = "${aws_launch_configuration.iam_profile_bug.id}"
  health_check_grace_period = 600
  health_check_type         = "EC2"

  tag {
    key = "Name"
    value = "${var.instance_name}"
    propagate_at_launch = true
  }

  tag {
    key = "Service"
    value = "${var.service_tag}"
    propagate_at_launch = true
  }
}

```
### Debug Output

If additional debug output is required I will provide based on request.
### Panic Output

NA
### Expected Behavior

When changing the IAM instance profile for EC2 Launch Configuration, Terraform should indicate a full replace of the instance is required
### Actual Behavior

A non-replacing update only indicator is provided.
### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2.  Uncomment the IAM instance profile for the launch configuration
3. `terraform plan`

```
Plan: 1 to add, 1 to change, 1 to destroy.
```
1. `terraform apply`

```
Apply complete! Resources: 1 added, 1 changed, 0 destroyed.
```
### Important Factoids

This is a custom built VPC and I did use dedicated tenancy during my tests, but I expect the same behavior. Perhaps more interesting is that if you start with the IAM profile associated and then remove the profile and attempt to apply:
1. `terraform plan`

```

~ aws_launch_configuration.iam_profile_bug
    iam_instance_profile: "eip_profile20160919215749035999894w4v" => ""


Plan: 0 to add, 1 to change, 0 to destroy.

```
1. `terraform apply`

```
aws_launch_configuration.iam_profile_bug: Modifying...
  iam_instance_profile: "eip_profile20160919215749035999894w4v" => ""
Error applying plan:

1 error(s) occurred:

* aws_launch_configuration.iam_profile_bug: doesn't support update

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```
### References

None

**Comment by moofish32** [comment]
Is this closed by 9f91c4e8c3712ca0a36cae624e55c4535aa0797a

**Comment by jen20** [comment]
Just tried to reproduce this with @moofish32 at the booth at reinvent - it's now fixed so this issue can be closed.
